src/ploneorg/addonlisting/utils.py

```diff
@@ -118,25 +131,42 @@ def update_addon(context, request=None):
             api.portal.show_message("Add'on %s has been updated" %
                                     (addon.title),
                                     request=request, type='info')
-            log.info(u'Finished to update: %s', addon.title)
+            if verbose:
+                log.info(u'Finished to update: %s', addon.title)
         else:
-            log.info(u'something went wrong on update %s', addon.title)
+            if verbose:
+                log.info(u'something went wrong on update %s', addon.title)
 
         if request is not None:
             request.response.redirect(context.absolute_url())
 
 
-def update_addons(context, request=None):
+def update_addons(context, request=None, verbose=False, limit=0):
     addon_folder = context
 
-    # get Add'on List
-    addons = api.content.find(context=addon_folder, portal_type="AddOn")
+    if limit:
```

Comment:
I couldn't understand.Are you referring to the fact that I used limit both as sort_limit parameter and as list slicing? That's because https://docs.plone.org/develop/plone/searching_and_indexing/query.html#sorting-and-limiting-the-number-of-results